pkg/skaffold/errors/err_def.go

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright 2019 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package errors
+
+import (
+	"fmt"
+
+	"github.com/GoogleContainerTools/skaffold/proto"
+)
+
+type Error interface {
+	Error() string
+	StatusCode() proto.StatusCode
+	Suggestions() []*proto.Suggestion
+}
+
+type ErrDef struct {
+	ae proto.ActionableErr
+}
+
+func (e ErrDef) Error() string {
+	if s := concatSuggestions(e.Suggestions()); s != "" {
+		return fmt.Sprintf("%s. %s", e.ae.Message, concatSuggestions(e.Suggestions()))
+	}
+	return e.ae.Message
+}
+
+func (e ErrDef) StatusCode() proto.StatusCode {
+	return e.ae.ErrCode
+}
+
+func (e ErrDef) Suggestions() []*proto.Suggestion {
+	return e.ae.Suggestions
+}
+
+func NewError(ae proto.ActionableErr) ErrDef {
```

Comment:
We should have a variant that takes and retains the wrapped `error`.  And we should implement `Unwrap() error` too.